horizomer/misc/taxonomy/tree_by_taxdump.py

```diff
@@ -0,0 +1,142 @@
+# this script generates a tree of a set genomes based on their NCBI taxonomy
+# (taxdump) annotations
+# usage: python me.py nodes.dmp genome2taxid.tsv output.nwk
+
+import sys
+from skbio import TreeNode
+
+# read NCBI taxonomy hierarchies (nodes.dmp)
```

Comment:
Another option: there is code in ETE3 to do a lot of this. You can, for instance, pass a list of NCBI taxids and [return the minimal tree](http://etetoolkit.org/docs/latest/tutorial/tutorial_ncbitaxonomy.html#getting-ncbi-species-tree-topology)  encompassing these taxids, eliminating the whole tree-building-pruning portion below. I believe ETE3 also now has a to_skbio method for tranlating to an skbio TreeNode object. 

It should then be simple to add genome IDs on as children to their appropriate taxids.

Reply:
![screenshot](https://user-images.githubusercontent.com/8292608/34846461-1dcadb7a-f6cd-11e7-8369-75068fad40a9.png)

Reply:
I believe that reproduces the functionality.

Reply:
It would be fantastic to finally get the pycogent code ([entry point](https://github.com/pycogent/pycogent/blob/master/cogent/parse/ncbi_taxonomy.py#L251)) for this ported

Reply:
@tanaes and @wasade , thanks for pointing me to the existing solutions!

I explored the **ETE** solution. It simply downloads the entire taxdump.tar.gz and builds an sqlite database under my home directory. This behavior is bit too invasive and inflexible. (e.g., I cannot plug my own nodes.dmp, nor can I do unit-tests easily.)

The **pycogent** code Daniel shared appear to be a lightweight and more reasonable solution. Certain codes may need to be updated (e.g., the RankNumbers are hard coded). I think it will be ideal to port it into scikit-bio or other general libraries. For now, let me just do the simplest homebrew way.